runtime/src/append_vec.rs

```diff
@@ -174,14 +208,27 @@ impl AppendVec {
 
     #[allow(clippy::mutex_atomic)]
     pub fn set_file<P: AsRef<Path>>(&mut self, path: P) -> io::Result<()> {
-        self.path = path.as_ref().to_path_buf();
         let data = OpenOptions::new()
             .read(true)
             .write(true)
             .create(false)
             .open(&path)?;
 
+        // this AppendVec must not hold actual file;
+        assert_eq!(self.file_size, 0);
+
+        let current_len = self.current_len.load(Ordering::Relaxed);
+        assert_eq!(current_len, *self.append_offset.lock().unwrap());
+
+        let file_size = std::fs::metadata(&path)?.len();
+        assert!(current_len <= file_size as usize);
+
+        AppendVec::sanitize_mmap_size(file_size as usize)?;
```

Comment:
Notice the `?`; Now we're correctly propagating this error up to the snapshot deserializing code.